File: 1.TwoSum.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{2, 2, 9, 11}, 4))
	fmt.Println(twoSum([]int{2, 1, 9, 11}, 3))
}

func twoSum(nums []int, target int) []int {
	index := make(map[int]int, len(nums))
	for i, v := range nums {
		if j, ok := index[target-v]; ok {
			return []int{j, i}
		}
		index[v] = i
	}
	return nil
}

File: 1.TwoSum.py
from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        data: List[int]
        index: Dict[int, int] = dict()
        for i, v in enumerate(nums):
            if target - v in index.keys():
                data = [index[target - v], i]
            index[v] = i
        return data


if __name__ == "__main__":
    s = Solution()
    print(s.twoSum([2, 2, 3, 4], 4))
    print(s.twoSum([2, 3, 3, 4], 6))
